### PR TITLE
Implement IJEPA Adapter

### LINKED ISSUE #39: [FEATURE] Implement I-JEPA Adapter for World Model Lens
## Is your feature request related to a problem?
The library currently lacks a native I-JEPA backend. While the `examples/ijepa/` directory in the `ui` branch contains an implementation of the ViT backbone and the I-JEPA structure (context encoder, predictor, target encoder), these components are not currently integrated into the core `WorldModelLens` framework. 

Specifically, there is no `WorldModelAdapter` for I-JEPA, which prevents standardized causal interpretability analysis and feature attribution using the unified `HookedWorldModel` API

## Describe the solution you'd like
Implementation of a new backend adapter in `world_model_lens/backends/ijepa_adapter.py` that maps the I-JEPA architecture onto the `WorldModelAdapter` interface. This involves:

1. **Mapping Architecture Components**:
   - `encode(obs)` : map to the I-JEPA `context_encoder` forward pass.
   - `dynamics(state)` : map to the I-JEPA `predictor` transformer forward pass.
2. **Implementing Weight Loader**: 
   - Add a `from_checkpoint()` implementation that supports loading official Meta weights, specifically the **ViT-H/14** backbone (`vith14_in22k_ep300.pth`).
3. **Framework Registration**: 
   - Use the `@register` decorator to add "ijepa" to the `BACKEND_REGISTRY` under the `WorldModelFamily.IJEPA` category.
4. **Consistency**:
   - Ensure it works seamlessly with the existing `HookedWorldModel` for automated hook extraction across the encoder and predictor blocks.

## Describe alternatives you've considered
- **Standalone Scripts**: In the `ui` branch, we have used the logic in `examples/ijepa/` as standalone scripts, but this lacks the framework-wide benefits of unified hook points, visualization suites, and standard causal evaluators.
- **Generic Transformer Adapter**: While possible, I-JEPA's unique masking strategy and the split between the context encoder and the predictor are distinct enough from autoregressive world models to benefit from a specialized, performant adapter.

## Additional context
This is a **Phase 0 (Blocker)** task for enabling I-JEPA support. The following foundational work is already present in the repository (ui branch):
- `examples/ijepa/vit.py`: Vision Transformer with exposed `forward_blocks`.
- `examples/ijepa/ijepa_model.py`: Core logic for the dual-encoder architecture.
- `examples/ijepa/causal_evaluator.py`: Initial intervention-based faithfulness tests.

### PR DESCRIPTION
**Pull request by skar07**
## Summary

Closes #39:  I-JEPA adapter implementation

### I-JEPA Backend (`world_model_lens/backends/ijepa_adapter.py`)
- **`IJEPAAdapter`**: Full implementation of the 3-component architecture:
    - **Context Encoder**: A VisionTransformer processing only visible patches (mapped to `encode()`).
    - **Target Encoder**: An EMA-updated (momentum) copy of the context encoder providing stable ground-truth targets.
    - **Predictor**: A transformer-based bottleneck that predicts target patch latents from context latents (mapped to `dynamics()`).
- **Structured Masking**: Implemented "Rect-on-Rect" strategy that samples 4 independent target blocks (~15-20% area each) and a large context block (~85% area) with resampling fallbacks to ensure minimum context retention.
- **Official Weight Support**: `from_checkpoint()` now maps official Meta ViT-H/14 weights (e.g. `vitH14_in22k_ep300.pth`) into the adapter's split-encoder structure by detecting `model` or `encoder` key prefixes.

### Core Integration
- **`HookedWorldModel`**: 
    - Added support for a dedicated `target_encoding` hook point. Captures the output of the target encoder at every step, enabling users to hook and compare predictions vs. actual ground-truth representations in the `ActivationCache`.
    - Fixed a critical registry bug in `from_checkpoint` that prevented dynamic lookup of newly registered backends.
- **Config**: Added `ijepa` backend and explicit control over predictor architecture (`predictor_embed_dim`, `predictor_depth`, `predictor_heads`).

### Verification & Examples
- **`examples/ijepa/train_ijepa.py`**: A complete mini-training script demonstrating the I-JEPA lifecycle (Masking -> Forward -> Block-wise MSE Loss -> Optimizer Step -> EMA Update).

## Testing

```bash
# Unit tests covering architectural init, masking logic, and EMA math
python -m pytest tests/backends/test_ijepa_adapter.py

# Gradient Isolation test (Ensure target encoder receives no grads)
python -m pytest tests/backends/test_ijepa_adapter.py::test_target_encoder_receives_no_gradients

# Integration test (Verifies HookedWorldModel compatibility and Meta weight mapping)
python -m pytest tests/backends/test_ijepa_integration.py
```

### CODE REVIEW

**Comment by Rishik00** [comment]
LGTM. Only suggestion: moving forward let's separate adapter updates and surrounding commits into different PRs.
